projects/addon-doc/directives/scroll-into-view/scroll-into-view.directive.ts

```diff
@@ -1,43 +1,33 @@
-import {Directive, ElementRef, Inject, Input, Self} from '@angular/core';
+import {Directive, ElementRef, inject, Input} from '@angular/core';
 import {TUI_DOC_PAGE_LOADED} from '@taiga-ui/addon-doc/tokens';
 import {TuiDestroyService, tuiGetElementObscures} from '@taiga-ui/cdk';
-import {
-    debounceTime,
-    filter,
-    Observable,
-    ReplaySubject,
-    switchMap,
-    takeUntil,
-} from 'rxjs';
+import {debounceTime, filter, ReplaySubject, switchMap, takeUntil} from 'rxjs';
 
 @Directive({
     selector: '[tuiScrollIntoViewLink]',
     providers: [TuiDestroyService],
 })
 export class TuiScrollIntoViewLinkDirective {
+    private readonly scroll$ = new ReplaySubject<boolean>(1);
+
     @Input()
     set tuiScrollIntoViewLink(shallWe: boolean) {
         this.scroll$.next(shallWe);
     }
 
-    private readonly scroll$ = new ReplaySubject<boolean>(1);
+    readonly readyToScroll$ = inject(TUI_DOC_PAGE_LOADED);
```

Comment:
Could we use it directly inside `constructor`?

```ts
inject(TUI_DOC_PAGE_LOADED).pipe(...)
```

Reply:
fixed